handlers/deltaValidationHandler_test.go

```diff
@@ -0,0 +1,163 @@
+)
+
+// TestNewDeltaValidationHandler asserts that the constructor for the DeltaValidationHandler returns a fully configured handler.
+func TestNewDeltaValidationHandler(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	Convey("When I call the constructor, then it returns me a valid DeltaValidationHandler", t, func() {
+
+		h := hMocks.NewMockHelper(mockCtrl)
+		chv := chvMocks.NewMockCHValidator(mockCtrl)
+
+		config.CallValidateConfig = func(cfg *config.Config) error {
+			return nil
+		}
+		cfg, _ := config.Get()
+
+		deltaValidationHandler := NewDeltaValidationHandler(h, chv, cfg)
+
+		So(deltaValidationHandler, ShouldNotBeNil)
+
+		So(deltaValidationHandler.chv, ShouldNotBeNil)
+		So(deltaValidationHandler.chv, ShouldEqual, chv)
+
+		So(deltaValidationHandler.h, ShouldNotBeNil)
+		So(deltaValidationHandler.h, ShouldEqual, h)
+	})
+}
+
+// TestDeltaValidationHandlerErrorsCallingValidation asserts that the DeltaValidationHandler returns an internal error status when
+// call to validate the request fails (internal failure such as failure to open schema, not a user validation failure).
+func TestDeltaValidationHandlerErrorsCallingValidation(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	Convey("Given a HTTP POST request via the officer delta validate endpoint", t, func() {
+
+		req := httptest.NewRequest(common.PostRestMethod, officersValidateEndpoint, bytes.NewBuffer([]byte(common.TestRequestBody)))
+		resp := httptest.NewRecorder()
+
+		Convey("When the request is handled by the router, but the kafka service fails to send", func() {
+
+			h := hMocks.NewMockHelper(mockCtrl)
+			chv := chvMocks.NewMockCHValidator(mockCtrl)
+
+			common.InitEnv()
+			config.CallValidateConfig = func(cfg *config.Config) error {
+				return nil
+			}
+			cfg, _ := config.Get()
+
+			handler := NewDeltaValidationHandler(h, chv, cfg)
+
+			h.EXPECT().GetRequestIdFromHeader(req).Return(common.TestContextId)
+			chv.EXPECT().ValidateRequestAgainstOpenApiSpec(req, handler.cfg.OpenApiSpec, common.TestContextId).Return(nil, errors.New("error"))
+
+			handler.ServeHTTP(resp, req)
+
+			Convey("Then the response should be 500 and an error returned", func() {
+				So(resp.Code, ShouldEqual, http.StatusInternalServerError)
+			})
+
+			common.DestroyEnv()
+		})
+	})
+}
+
+// TestDeltaValidationHandlerSuccessfulValidation asserts that the DeltaValidationHandler returns an ok status when validation passes
+func TestDeltaValidationHandlerSuccessfulValidation(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	Convey("Given a HTTP POST request via the officer delta validate endpoint", t, func() {
+
+		req := httptest.NewRequest(common.PostRestMethod, officersValidateEndpoint, bytes.NewBuffer([]byte(common.TestRequestBody)))
+		resp := httptest.NewRecorder()
+
+		Convey("When the request is handled by the router, but the kafka service fails to send", func() {
```

Comment:
Same as before, I think the Convey for these tests will need a slight rework as they seem outdated (kafka mentioned)